internal/manager/config.go

```diff
@@ -0,0 +1,51 @@
+package manager
+
+import (
+	"flag"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	LeaderElectionDefault         = true
+	MetricsBindAddressDefault     = ":8080"
+	HealthProbeBindAddressDefault = ":8081"
+	SyncPeriodDefault             = 3 * time.Hour
+)
+
+type Config struct {
+	Namespaces             []string
+	EnableLeaderElection   bool
+	MetricsBindAddress     string
+	HealthProbeBindAddress string
+	SyncPeriod             time.Duration
+}
+
+func (c *Config) ParseWatchedNamespaces() {
+	// dev note: we pass the watched namespace as the env variable to use the Kubernetes Downward API. Unfortunately
+	// there is no way to use it for container arguments
+	watchedNamespace := os.Getenv("WATCH_NAMESPACE")
+
+	c.Namespaces = make([]string, 0, 1)
+
+	for _, namespace := range strings.Split(watchedNamespace, ",") {
+		c.Namespaces = append(c.Namespaces, strings.TrimSpace(namespace))
+	}
+}
```

Comment:
The non main code should not manage OS env vars or CLI flags, those are main CLI concepts.

I would move this to the main, but leave the config struct definition here.